server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -791,6 +798,50 @@ public void run() {
       }
     }, 0, 5000, TimeUnit.MILLISECONDS);
 
+    // Periodically check that metadata of tablets matches what is held in memory
+    ThreadPools.createGeneralScheduledExecutorService(aconf).scheduleWithFixedDelay(() -> {
+      final SortedMap<KeyExtent,Tablet> onlineTabletsSnapshot = onlineTablets.snapshot();
+
+      final SortedSet<KeyExtent> userTablets = new TreeSet<>();
+      final SortedSet<KeyExtent> nonUserTablets = new TreeSet<>();
+
+      // Create subsets of tablets based on DataLevel: one set who's DataLevel is USER and another
+      // containing the remaining tablets (those who's DataLevel is ROOT or METADATA).
+      // This needs to happen so we can use .readTablets() on the DataLevel.USER tablets in order
+      // to reduce RPCs.
+      // TODO: Push this partitioning, based on DataLevel, to ample.
+      onlineTabletsSnapshot.forEach((k, v) -> {
+        if (Ample.DataLevel.of(k.tableId()) == Ample.DataLevel.USER) {
+          userTablets.add(k);
+        } else {
+          nonUserTablets.add(k);
+        }
+      });
+
+      List<TabletMetadata> tmdList;
+
```

Comment:
should get the update counts before reading from the metadata table.  This allows to detect changes that happen while reading the metadata.

```suggestion
Map<KeyExtent, Long> updateCounts = new HashMap<>()
onlineTabletsSnapshot.forEach((ke,tablet) -> {
   updateCounts.put(ke, tablet.getUpdateCount());
});
```